analyses/cellranger-analysis/j1.sh

```diff
@@ -12,17 +12,36 @@ module load cellranger/8.0.1
 # Set up running directory
 cd "$(dirname "${BASH_SOURCE[0]}")" 
 
+########################################################################
+# Read multiple values and assign them to variables by parsing yaml file
+metadata_dir=$(cat ../../project_parameters.Config.yaml | grep 'metadata_dir:' | awk '{print $2}')
+metadata_dir=${metadata_dir//\"/}  # Removes all double quotes
+echo "$metadata_dir"  # Output: This is a string with quotes.
+
+genome_reference_path=$(cat ../../project_parameters.Config.yaml | grep 'genome_reference_path:' | awk '{print $2}')
+genome_reference_path=${genome_reference_path//\"/}  # Removes all double quotes
+echo "$genome_reference_path"  # Output: This is a string with quotes.
+
+cellranger_parameters=$(cat ../../project_parameters.Config.yaml | grep 'cellranger_parameters:' | awk '{print $2}')
+cellranger_parameters=${cellranger_parameters//\"/}  # Removes all double quotes
+echo "$cellranger_parameters"  # Output: This is a string with quotes.
+
+########################################################################
+# Create directories to save output files to
+mkdir -p ./results/01_logs
+mkdir -p ./results/02_cellranger_count
+mkdir -p ./results/02_cellranger_count/${cellranger_parameters}
+echo "./results/02_cellranger_count/${cellranger_parameters}"
+
+########################################################################
 # If your `project_metadata` is not in `*.txt` file format
 # use the following code line to convert it
-cat ./project_metadata.tsv | sed 's/,/\t/g' > ./input/project_metadata.txt
-
+cat "${metadata_dir}"/project_metadata.tsv | sed 's/,/\t/g' > ./input/project_metadata.txt
 
 ########################################################################
 # Run CellRanger for all libraries
 python ./util/run_cellranger.py --file=./input/project_metadata.txt \
-                                --transcriptome=GRCm39 \
+                                --transcriptome=${genome_reference_path}  \
                                 --create_bam=true \
-                                --output_dir=./results/02_cellranger_count/DefaultParameters/
-                                # --force_cells=8000
-                                # --output_dir=./results/02_cellranger_count/ForcedCells8000Parameters/
-                                
\ No newline at end of file
+                                --output_dir=./results/02_cellranger_count/${cellranger_parameters}/
```

Comment:
Small suggestion, consider including \ after output_dir path, so if --force_cells parameter is used, the user won't need to worry about adding the \ so the --force_cells parameter will be recognized

Reply:
```suggestion
                                --output_dir=./results/02_cellranger_count/${cellranger_parameters}/ \
```

Reply:
The inclusion of `\` leads to breaking the script, unfortunately. How about removing the flag altogether?

Reply:
Yeah, I suppose it could be removed, since it most likely won't be commonly used. Or if you want to leave, maybe add an instruction to the README that if --force_cells flag is used, add \ to the end of the line above or something like that? Or maybe that would be unnecessarily confusing for users and it's better to remove...